File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/model/ParsedWorkoutResult.kt
package com.scribblefit.feature.ai.domain.model

data class ParsedWorkoutResult(
    val workout: ParsedWorkout?,
    val rawText: String,
    val status: ParsingStatus,
    val modelUsed: String? = null,
    val processingTimeMs: Long = 0,
    val reasoning: String? = null,
    val error: String? = null
)

enum class ParsingStatus { SUCCESS, PARTIAL_SUCCESS, FAILURE }

File: apps/ios/ScribbleFit/ScribbleFit/Core/AI/ParsedWorkoutResult.swift
import Foundation

/**
 * Standardized result for workout parsing with rich metadata.
 */
public struct ParsedWorkoutResult: Sendable {
    public let workout: ParsedWorkout?
    public let rawText: String
    public let status: ParsingStatus
    public let modelUsed: String?
    public let processingTimeMs: Int64
    public let reasoning: String?
    public let error: String?

    public init(
        workout: ParsedWorkout?,
        rawText: String,
        status: ParsingStatus,
        modelUsed: String? = nil,
        processingTimeMs: Int64 = 0,
        reasoning: String? = nil,
        error: String? = nil
    ) {
        self.workout = workout
        self.rawText = rawText
        self.status = status
        self.modelUsed = modelUsed
        self.processingTimeMs = processingTimeMs
        self.reasoning = reasoning
        self.error = error
    }
}

public enum ParsingStatus: String, Codable, Sendable {
    case success = "SUCCESS"
    case partialSuccess = "PARTIAL_SUCCESS"
    case failure = "FAILURE"
}
